recipes/pcapplusplus/all/conanfile.py

```diff
@@ -35,37 +41,56 @@ def requirements(self):
         else:
             self.requires("libpcap/1.9.1")
 
+    def validate(self):
+        if self.settings.os not in ("FreeBSD", "Linux", "Macos"):
+            raise ConanInvalidConfiguration("%s is not supported" % self.settings.os)
+
     def source(self):
-        tools.get(
-            **self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, 
-            strip_root=True,
-        )
+        tools.get(**self.conan_data["sources"][self.version],
+            destination=self._source_subfolder, strip_root=True)
+
+    @property
+    def _configure_sh_script(self):
+        return {
+            "Android": "configure-android.sh",
+            "Linux": "configure-linux.sh",
+            "Macos": "configure-mac_os_x.sh",
+            "FreeBSD": "configure-freebsd.sh",
+        }[str(self.settings.os)]
+
+    def _patch_sources(self):
+        if not self.options.get_safe("fPIC"):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "PcapPlusPlus.mk.common"),
+                                  "-fPIC", "")
```

Comment:
Do we need to support builds without `fPIC`?

Reply:
It's not an absolute requirement, but if it's possible without much work, why not?

Reply:
It's not very pretty to remove the `-fPIC` flag from `PcapPlusPlus.mk.common`, so if we can avoid it I think we should consider it. Let me know what you think

Reply:
I think the "prettyness" is acceptable.
Alternatively, we can add to the makefiles such that pcapplusplus can optionally be built with fpic support.

Then, we can pass `FPIC=1` for fpic enabled and `FPIC=0` for fpic disabled.
Also, this approach won't change the current default (enabling fpic)
e.g.
```makefile
FPIC=1

if FPIC
PCAPPP_BUILD_FLAGS += -fPIC
endif
```

Reply:
I can do that for future versions, but I'm not sure if there is a need to build without `-fPIC` 🤔

Reply:
@madebr let's decide on this and merge your PR?

Reply:
Great, then the other members of conan-center-index can review too.

Reply:
@madebr So do you prefer we keep supporting builds without `-fPIC`?

Reply:
I would keep it.
Unless you will move the location where the fpic flag is added, it just works now.